deadman

```diff
@@ -247,6 +254,39 @@ class Ping :
 
         return res
 
+    def sendSnmpPing (self) :
+        if not self.relay.has_key("community") :
+            raise RuntimeError ("\"community\" is not specified for %s" %
+                                self.addr)
+        community = self.relay["community"]
+        community = community.replace("\\", "\\\\")
+        community = community.replace("'", "\\'")
+        snmpcmd = "snmpping -Cc1 -v 2c -c \'%s\' " % community
+        if not self.relay.has_key("relay") :
+            raise RuntimeError ("\"relay\" is not specified for %s" %
+                                self.addr)
+        snmpcmd += " %s " % self.relay["relay"]
+
+        pingcmd = snmpcmd + " %s" % self.addr
+
+        result = commands.getoutput (pingcmd)
+
+        rttm = re.search (r'rtt min/avg/max/stddev = \d+', result)
+
+        res = PingResult ()
+
+        if rttm :
+            res.success = True
+            res.errcode = PING_SUCCESS
+            res.rtt = float (rttm.group (0).split ('=')[1])
```

Comment:
If you make your regexp

`r'rtt min/avg/max/stddev = (\d+)'`

(i.e., add parenthesis around the \d+)

then this line can simply contain `float( rttm.group( 1 ) )`.